### PR TITLE
Update some modules documentation

### PR DESCRIPTION
**Pull request by SergioGasquez**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
Updated the module documentation of I2C, LEDC and UART. Documents how to invert signals in UART (https://github.com/esp-rs/esp-hal/issues/1216)

#### Testing
I've run `run-doc-test` of `xtask`

### CODE REVIEW

**Review by bjoernQ — APPROVED** [review]
LGTM - Thanks!

File: esp-hal/src/uart.rs [hunk 1]
```diff
@@ -30,6 +30,10 @@
 //! # }
 //! ```
 //! 
+//! The UART controller can be configured to invert the polarity of the pins.
+//! This is achived by inverting the dessired pins, and then constucting the
```
Comment (jessebraham) [inline esp-hal/src/uart.rs:34]:
```suggestion
//! This is achived by inverting the desired pins, and then constucting the
```

Typo here

**Review by jessebraham — APPROVED** [review]
Thanks for this, looks good to me! I did just notice one small typo though, if you don't mind fixing it.

File: esp-hal/src/uart.rs (hunk 1 above)
Comment (jessebraham) [reply esp-hal/src/uart.rs:34]: I guess I can just fix it on GitHub, always forget about that 😅

**Pull request merged**